### PR TITLE
Makes inference endpoint the primary way to download and deploy ELSER and E5

### PR DESCRIPTION
**Pull request by szabosteve**
## Overview

This PR overhauls the ELSER and E5 conceptual pages:
* indicates that the recommended and easiest way to download and deploy the models is using inference endpoints,
* introduces the procedure of creating an inference endpoint in the `Download and deploy` section,
* makes the alternative download and deploy methods less prominent by putting them into collapsible sections that are closed by default,
* propagates the semantic_text workflow for semantic search.

### Preview

* [ELSER](https://stack-docs_bk_2765.docs-preview.app.elstc.co/guide/en/machine-learning/master/ml-nlp-elser.html)
* [E5](https://stack-docs_bk_2765.docs-preview.app.elstc.co/guide/en/machine-learning/master/ml-nlp-e5.html)

### CODE REVIEW

File: docs/en/stack/ml/nlp/ml-nlp-e5.asciidoc [hunk 1]
```diff
@@ -83,11 +121,15 @@ allocations and threads per allocation values.
 image::images/ml-nlp-deployment-id-e5.png[alt="Deploying ELSER",align="center"]
```
Comment (maxhniebergall) [inline docs/en/stack/ml/nlp/ml-nlp-e5.asciidoc:121]: not related to this PR but it seems that the alt text for this image is wrong

**Review by maxhniebergall — COMMENTED** [review]
This looks good, but I just want to double check with Jan that adaptive allocations made it into the 8.15 release before I approve this.

**Comment by szabosteve** [comment]
@maxhniebergall Thanks for the review! I just merged the adaptive allocations-related documentation yesterday. Jan asked me to draft the docs and he approved the PRs. You can find these here:
* [API docs](https://github.com/elastic/elasticsearch/pull/111476)
* [Conceptual docs](https://github.com/elastic/stack-docs/pull/2763)
* [tracking issue](https://github.com/elastic/ml-team/issues/1290)
However, I can change the API requests in this PR to NOT use adaptive_allocations if you prefer.

**Comment by szabosteve** [comment]
@maxhniebergall I replaced the `adaptive_allocations` settings with `num_of_allocations` and `num_of_threads` via [e11530e](https://github.com/elastic/stack-docs/pull/2765/commits/e11530e8bcde5a2b89eaff8f8dd1571f03f174c5)

**Review by maxhniebergall — APPROVED** [review]
Looks great! Thanks for the quick fixes. We will definitely want to re-add the adaptive allocations in 8.16.

**Pull request merged**